Fix for records with prim namespace prefix

jaronkk opened the pull request:
We're writing a search landing page that splits our local holdings and electronic results similar to what Villanova is doing: https://library.villanova.edu/Find/Resource/Results?lookfor=van+gogh&type=&search=allresources&submit=Find

When I attempted to search Primo Central I found that some records were not being parsed correctly due to being namespaced as prim:record.  This pull requests sends the original namespaces along with the raw_xml, then builds an xml document and removes the namespaces so it matches the expected structure of <record></record>.

I'm not tied to the way I implemented the fix so let me know if you'd like to see it done in a different manner or if the tests are lacking.

jaronkk committed:
Fix for parsing records with the prim namespace prefix.

scotdalton commented:
Jaron,
Awesome.  Thanks for the pull request.  I'll have a look tomorrow and let you know.  I don't have Primo Central, so I was just building off the Exlibris documentation.  Any sort of VCR cassette with a Primo Central response would be incredibly helpful.

Thanks again.

jaronkk commented:
When I implemented my fix I figured out how to use VCR and added a cassette of a Primo Central response.  I added tests to the XmlUtil test checking the namespace cleaning, and added a separate Primo Central test which makes use of the cassette.

Let me know if you have any question!

scotdalton commented:
Thanks again for the fix and VCR cassette.  I merged and pushed the latest to rubygems.
